### PR TITLE
Add geo shape query filter

### LINKED ISSUE #213: [FEATURE] Add filter by drawing boundary on Maps
### Is your feature request related to a problem?
Filter maps layer by drawing boundary manually on maps

### What solution would you like?
In Maps i should be able to filter documents by drawing boundary on maps page.

### What alternatives have you considered?
N/A

### Do you have any additional context?
N/A

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Added Geo Shape Query Filter to build geo shape query for given shape , relations and label.

### Issues Resolved
#213 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/model/geo/filter.ts [hunk 1]
```diff
@@ -4,9 +4,22 @@
...
+  FILTERS,
+  GeoBoundingBoxFilter,
+} from '../../../../../src/plugins/data/common';
 import { GeoBounds } from '../map/boundary';
 
+export type FilterRelations = 'INTERSECTS' | 'DISJOINT' | 'WITHIN';
+
+export type GeoShapeFilter = Filter & {
+  meta: FilterMeta;
+  geo_shape: any;
```
Comment (junqiu-lei) [inline public/model/geo/filter.ts:20]: Could we define the type of geo_shape?

File: public/model/layerRenderController.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ import {
   IOpenSearchDashboardsSearchResponse,
   isCompleteResponse,
   TimeRange,
-  Query,
+  Query, FILTERS,
```
Comment (junqiu-lei) [inline public/model/layerRenderController.ts:18]: nit: move FILTERS to next line.

File: public/model/geo/filter.ts (hunk 1 above)
Comment (VijayanB) [reply public/model/geo/filter.ts:20]: Geo shape has variable field name property and fixed string property. How do you recommend create type in that case?

File: public/model/layerRenderController.ts (hunk 2 above)
Comment (VijayanB) [reply public/model/layerRenderController.ts]: Ack

File: public/model/geo/filter.ts (hunk 1 above)
Comment (junqiu-lei) [reply public/model/geo/filter.ts:20]: Maybe we can define something like:

```
type GeoShape = {
  ignore_unmaped: Boolean,
  location: {
    relation: FilterRelations,
    shape: {
      type: 'Polygon',
      coordinates: [
        [
          [number, number],
          [number, number],
          [number, number],
          [number, number],
        ],
      ],
    },
  }
}
```

Comment (VijayanB) [reply public/model/geo/filter.ts:20]: but location is not fixed. It could be any field name

Comment (junqiu-lei) [reply public/model/geo/filter.ts:20]: In short term I think it's ok to use `any`, long term we need define the type to make it safer usage

Comment (junqiu-lei) [reply public/model/geo/filter.ts:20]: We might can use generic type for it?

```
type Coordinate = [number, number];

type Polygon = {
  type: 'Polygon';
  coordinates: Coordinate[][];
};

type Geometry = Polygon;

type Location<F extends string> = {
  [K in F]: {
    relation: 'INTERSECTS';
    shape: Geometry;
  };
};

type GeoShape<F extends string> = {
  ignore_unmaped: true;
} & Location<F> & Geometry;

```

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**